### PR TITLE
Remove /articles/server

### PR DESCRIPTION
**Pull request by kwalrath**
And move /articles/server/native-extensions to /server/c-interop-old.

I'm not sure I got the facts right, so I'll need a sanity check from mit-mit@.

### CODE REVIEW

File: src/server/c-interop-old.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-category: dart-vm
+description: The original way for command-line Dart apps to call C/C++ functions.
 ---
 
 _Written by William Hesse <br>
 May 2012_
 
 <aside class="alert alert-warning" markdown="1">
-  The example code on this page has some
+  The extension mechanism discussed in this page is being replaced by an
```
Comment (mit-mit) [inline src/server/c-interop-old.md]: I don't know that we have decided to do that, actually. Let's be a bit more general, e.g.:

The extension mechanism discussed in this page is for deep integration of the VM. If you just need to call existing code written in C or C++, see [C & C++ interop using FFI]()

File: src/server/c-interop.md [hunk 2]
```diff
@@ -12,4 +12,7 @@ to call native C and C++ APIs.
 This mechanism is currently [in active
 development](https://github.com/dart-lang/sdk/issues/34452) and is not yet
 complete. For a small preview example, see the [dart:ffi sqllite
-sample](https://github.com/dart-lang/sdk/blob/master/samples/ffi/sqlite/README.md).
+sample.](https://github.com/dart-lang/sdk/blob/master/samples/ffi/sqlite/README.md)
+
+For information on the old native interface, see
```
Comment (mit-mit) [inline src/server/c-interop.md]: I don't think we need to link in this direction

File: src/server/c-interop-old.md [hunk 3]
```diff
@@ -1,15 +1,15 @@
 ---
 title: Native extensions for the standalone Dart VM
-description: Learn how to enable command-line Dart apps to call C/C++ functions.
-date: 2012-05-01
-category: dart-vm
+description: The original way for command-line Dart apps to call C/C++ functions.
```
Comment (mit-mit) [inline src/server/c-interop-old.md]: I think I would use something like 'Integrating Dart and native code using native extensions'

Also, I would prefer not to rename the file. Old carries connotations. Native-extensions.md is neutral.

File: src/_data/side-nav.yml [hunk 4]
```diff
@@ -84,8 +84,6 @@
           permalink: /server/libraries
         - title: C & C++ interop
           permalink: /server/c-interop
-        - title: Native extensions
```
Comment (mit-mit) [inline src/_data/side-nav.yml]: I think we may want to keep (an updated version) of this page for the specialized use case of embedding Dart into existing native apps. Can we log a bug to remind ourselves to discuss that later?

File: src/server/c-interop-old.md (hunk 3 above)
Comment (kwalrath) [reply src/server/c-interop-old.md]: I'd like it to start with the same letters as the other page, so the other page will be selected in the nav when this one is current. Maybe... c-interop-extensions.

File: src/_data/side-nav.yml (hunk 4 above)
Comment (kwalrath) [reply src/_data/side-nav.yml]: Done: #1548

**Comment by kwalrath** [comment]
ptal

File: src/server/c-interop-old.md (hunk 3 above)
Comment (mit-mit) [reply src/server/c-interop-old.md]: `c-interop-native-extensions` would be ok

**Pull request merged**